framework/wazuh/agent.py

```diff
@@ -1519,48 +1519,59 @@ def create_multi_group(group_id):
 
 
     @staticmethod
-    def remove_multi_group(group_id):
+    def remove_multi_group(groups_id):
+        """
+        Removes groups by IDs.
 
-        if group_id.lower() == "default":
-                raise WazuhException(1712)
+        :param groups_id: list with Groups ID.
+        """
 
-        for filename in listdir("{0}".format(common.groups_path)):
-            file = open("{0}/{1}".format(common.groups_path,filename),"r")
-            agent_id = filename
-            agent_group = file.read()
-            agent_group = agent_group.strip()
-            file.close()
+        if not isinstance(groups_id, list):
+            raise WazuhException(1104, "Input must be a list")
 
-            if agent_group.find(group_id) >= 0:
+        for group in groups_id:
+            if group.lower() == "default":
+                raise WazuhException(1712)
 
-                group_list = agent_group.split(',')
-                # remove the group
-                group_list.remove(group_id)
-                if len(group_list) > 1:
-                    # create new multigroup
-                    new_group = ','.join(group_list)
-                    if not Agent.multi_group_exists(new_group):
-                        Agent.create_multi_group(new_group)
-                else:
-                    new_group = 'default' if not group_list else group_list[0]
-                
-                # Add multigroup
-                agent_file = open("{0}/{1}".format(common.groups_path,agent_id),"w")
-                agent_file.write("{0}\n".format(new_group))
-                agent_file.close()
+        for filename in listdir("{0}".format(common.groups_path)):
+            for group in groups_id:
+                file = open("{0}/{1}".format(common.groups_path, filename), "r")
+                agent_id = filename
+                agent_group = file.read()
+                agent_group = agent_group.strip()
```

Comment:
This can be shortened: `agent_group = file.read().strip()`. Also, it's not a good idea to call the variable `file` since it shadows `file` function in python.